explorer/app/components/ComponentCreator/ComponentCreator.tsx

```diff
@@ -0,0 +1,36 @@
+import { ComponentObject } from "app/config/model";
+import { DetailResponseType } from "app/models/responses";
+import React from "react";
+
+interface ComponentCreatorProps {
+  components: ComponentObject[];
+  detail: DetailResponseType;
+  deep?: number;
+}
+
+export const ComponentCreator = ({
```

Comment:
Should we call this `DetailComponentCreator` as it looks like this is specified to detail page functionality? Or should we keep it generic and rename/retype the `detail: DetailResponseType` prop to `responseType: AppropriateTypeHere`? We can discuss on our call this afternoon.

Reply:
As discussed, `responseType: T`.